### PR TITLE
Removed memory retain cycle for block

### PR DESCRIPTION
**Pull request by bhabani-sixt**
Block, copy to heap and create memory retain cycle. Added weak to break retain cycle chain.

### CODE REVIEW

File: Swen/Classes/Swen.swift [hunk 1]
```diff
@@ -123,8 +123,8 @@ fileprivate extension Swen where EventType: StickyEvent {
         let listener = EventListener<EventType>(observer, queue, handler)
         listeners.append(listener)
         if let sticky = sticky {
-            listener.queue.addOperation {
-                listener.post(sticky)
+            listener.queue.addOperation {[weak listener] in
```
Comment (kimlanbui) [inline Swen/Classes/Swen.swift]: there is a missing space between { and [

File: Swen/Classes/Swen.swift [hunk 2]
```diff
@@ -192,8 +192,8 @@ fileprivate class EventListener<EventType: BaseEvent> {
         if OperationQueue.current == queue {
             handler(event)
         } else {
-            queue.addOperation {
-                self.handler(event)
+            queue.addOperation {[weak self] in
```
Comment (kimlanbui) [inline Swen/Classes/Swen.swift]: there is a missing space between { and [

File: Example/Tests/SwenTests.swift [hunk 3]
```diff
@@ -232,4 +232,23 @@ class SwenTests: XCTestCase {
         Swen.post(TestEvent())
     }
 
+    
+    func testReleaseListeners() {
```
Comment (dmitryPoznukhov) [inline Example/Tests/SwenTests.swift:236]: actually I think, it's enough to have:

   ```
 func testReleaseListeners() {
        Swen.post(TestStickyEvent())

        Swen<TestStickyEvent>.register(self) { _ in }
        weak var listener = Swen<TestStickyEvent>.instance(in: .defaultStorage).listeners.first
        Swen<TestStickyEvent>.unregister(self)

        XCTAssertNil(listener)
    }
```

File: Swen/Classes/Swen.swift [hunk 4]
```diff
@@ -192,8 +192,8 @@ fileprivate class EventListener<EventType: BaseEvent> {
         if OperationQueue.current == queue {
             handler(event)
         } else {
-            queue.addOperation {
```
Comment (dmitryPoznukhov) [inline Swen/Classes/Swen.swift:195]: please, add test, to show that this changes is necessary

**Review by dmitryPoznukhov — CHANGES_REQUESTED** [review]

File: Example/Tests/SwenTests.swift (hunk 3 above)
Comment (bhabani-sixt) [reply Example/Tests/SwenTests.swift:236]: +1

File: Swen/Classes/Swen.swift (hunk 4 above)
Comment (bhabani-sixt) [reply Swen/Classes/Swen.swift:195]: Added new test cases.

File: Example/Tests/SwenTests.swift [hunk 5]
```diff
@@ -232,4 +232,49 @@ class SwenTests: XCTestCase {
...
+            weak var listener = Swen<TestEvent>.instance(in: .defaultStorage).listeners.first
+            Swen<TestEvent>.unregister(self)
+            XCTAssertNil(listener)
+            exp.fulfill()
+        }
+        
+        waitForExpectations(timeout: timeout)
+    }
+    
+    func testReleaseListenersForStickyEvent() {
```
Comment (dmitryPoznukhov) [inline Example/Tests/SwenTests.swift]: you can do it in this way as well:

```
    func testReleaseListenersForNonStickyEvent() {
        let registerQueue = OperationQueue()

        Swen<TestEvent>.register(self, onQueue: registerQueue) { _ in }
        weak var listener = Swen<TestEvent>.instance(in: .defaultStorage).listeners.first
        Swen.post(TestEvent())
        Swen<TestEvent>.unregister(self)

        XCTAssertNil(listener)
    }
```

**Review by dmitryPoznukhov — CHANGES_REQUESTED** [review]

**Review by dmitryPoznukhov — APPROVED** [review]

**Pull request merged**